tasks/helpers/getPackagePaths.js

```diff
@@ -41,23 +37,36 @@ var recursivelyAccumulateRequiredPaths = function(accumPaths, hostPath) {
     // Add the paths to our list and recursively search their associated files.
     .reduce(function(accumPaths, relativeRequiredPath) {
       var normalizedRequiredPath = path.join(path.dirname(hostPath), relativeRequiredPath);
-      return recursivelyAccumulateRequiredPaths(accumPaths, normalizedRequiredPath);
+
+      // only process the require statements for a file we haven't seen yet to avoid circular imports
+      if (accumPaths.indexOf(normalizedRequiredPath) === -1) {
+        return recursivelyAccumulateRequiredPaths(accumPaths, normalizedRequiredPath);
+      }
+
+      return accumPaths;
     }, accumPaths);
   return accumPaths;
 };
 
+var ensureArray = function (value) {
```

Comment:
`['hello'].concat([]) = ['hello']`

`['hello'].concat(['world']) = ['hello', 'world']`

This way, `getLibPaths` can always call `.concat` without worry. we'll return an empty array when the incoming `value` is exists.